sssd_test_framework/roles/samba.py

```diff
@@ -1172,6 +1202,258 @@ def lockout(self, duration: int, attempts: int) -> SambaPasswordPolicy:
+        """
+        Get DNS record
+
+        :param fqname: Fully qualified hostname.
+        :type fqname: str
+        :param ipv6: IPv6 switch, optional
+        :type ipv6: bool | None = False
+        :return: Parsed a/aaaa record, soa record
+        :rtype: tuple[dict | None, Any | None]
+        """
+        record_type = "AAAA" if ipv6 else "A"
+        result = self.host.conn.run(f"dig {fqname} {record_type}").stdout
+        return jc_parse_dig(result)
+
+    def get_ptr_record(self, ip: str) -> tuple[dict | None, Any | None]:
+        """
+        Get DNS ptr or soa record data.
+
+        :param ip: IP address.
+        :type ip: str
+        :return: Parsed ptr record, soa record.
+        :rtype:  tuple[dict | None, Any | None]
+        """
+        result = self.host.conn.run(f"dig -x {ip}").stdout
+        return jc_parse_dig(result)
+
+
+class SambaDNSZone(SambaDNSServer):
+    """
+    DNS zone management.
+    """
+
+    def __init__(self, role: Samba, name: str):
+        """
+        :param role: Samba host object.
+        :type role: SambaHost
+        :param name: DNS zone name.
+        :type name: str
+        """
+        super().__init__(role)
+
+        self.zone_name: str = name
+        """Zone name."""
+
+    def create(self) -> SambaDNSZone:
+        """
+        Create new zone.
+
+        :return: SambaDNSServer object.
+        :rtype: SambaDNSServer
+        """
+        self.host.conn.run(f"samba-tool dns zonecreate {self.server} {self.zone_name} {self.credentials}")
+        return self
+
+    def delete(self) -> None:
+        """
+        Delete zone.
+        """
+        self.host.conn.run(f"samba-tool dns zonedelete {self.server} {self.zone_name} {self.credentials}")
+
+    def add_record(self, shortname: str, ip: str, ipv6: bool | None = False) -> SambaDNSZone:
+        """
+        Add DNS record.
+
+        :param shortname: Short hostname.
+        :type shortname: str
+        :param ip: IP address.
+        :type ip: str
+        :param ipv6: IPv6 switch, optional
+        :type ipv6: bool | None = False
+        :return: SambaDNSZone object.
+        :rtype: SambaDNSZone
+        """
+        record_type = "AAAA" if ipv6 else "A"
+
+        self.host.conn.run(
+            f"samba-tool dns add "
+            f"{self.server} {self.zone_name} "
+            f"{shortname} {record_type} {ip} "
+            f"{self.credentials}"
+        )
+        return self
+
+    def add_ptr_record(self, fqname: str, ip: str) -> SambaDNSZone:
+        """
+        Add DNS ptr record.
+        :param fqname: Fully qualified hostname.
+        :type fqname: str
+        :param ip: IP address.
+        :type ip: str
+        :return: SambaDNSZone object.
+        :rtype: SambaDNSZone
+        """
+        self.host.conn.run(
+            f"samba-tool dns add " f"{self.server} {self.zone_name} " f"{ip} PTR {fqname} " f"{self.credentials}"
```

Comment:
Please format this similar to `add_record` above with newlines.

Reply:
Yea, I don't know what happened there.